src/stringfilter.cpp

```diff
@@ -119,12 +110,12 @@ void StringFilter::AddLine(const char *str)
 	for (WordState &ws : this->word_index) {
 		if (!ws.match) {
 			if (this->locale_aware) {
-				if (match_case ? StrNaturalContains(str, ws.start) : StrNaturalContainsIgnoreCase(str, ws.start)) {
+				if (match_case ? StrNaturalContains(str, ws.word) : StrNaturalContainsIgnoreCase(str, ws.word)) {
 					ws.match = true;
 					this->word_matches++;
 				}
 			} else {
-				if ((match_case ? strstr(str, ws.start) : strcasestr(str, ws.start)) != nullptr) {
+				if ((match_case ? strstr(str, ws.word.c_str()) : strcasestr(str, ws.word.c_str())) != nullptr) {
```

Comment:
Shame about `.c_str()` here. I guess `.find()` could work to replace the `strstr()` call, but without an equivalent for `strcasestr()`, that's a bit confuseling.